### PR TITLE
Task monitoring

### PR DESCRIPTION
**Pull request by harnash**
Added automatic status update of  tasks running in Celery to properly reflect failures/success in DB.
Added option to allow filtering/searching through API.
There is an issue which needs to be resolved and it is the matter of sharing some common parts between Django App and Celery workers (TaskStatus, etc.) but this is a bigger thing and it needs some research and discussion in larger group.
Did some minor refactor here and there. Also removed some legacy task_runner code.

### CODE REVIEW

File: apps/testrunner/monitor.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+from celery.utils.log import get_task_logger
+
+logger = get_task_logger(__name__)
+
+def set_task_status(task, status):
+    if task is None or not task.request.kwargs:
+        return
+
+    # only report success on final task and only one with proper key
+    if status == 2 or 'task_uri' not in task.request.kwargs:
```
Comment (wladekb) [inline apps/testrunner/monitor.py]: `or status == -1` ?

File: apps/testrunner/test_suites/simple.py [hunk 2]
```diff
@@ -34,19 +35,37 @@ def run(self, retries, **kwargs):
...
+                    task_uri=kwargs['task_uri'],
+                ),
+            ),
             ProcessResponses().s(
                 result_uri=kwargs['result_uri'],
+                test_run_uri=kwargs['test_run_uri'],
                 task_uri=kwargs['task_uri'],
-                test_run_uri=kwargs['test_run_uri']
-            )
+                final_task=True,
```
Comment (wladekb) [inline apps/testrunner/test_suites/simple.py]: it doesn't seem to be used anywhere...

File: apps/testrunner/monitor.py (hunk 1 above)
Comment (harnash) [inline apps/testrunner/monitor.py]: Right. That makes sense.

File: apps/testrunner/test_suites/simple.py (hunk 2 above)
Comment (harnash) [inline apps/testrunner/test_suites/simple.py]: Yes. I did forget to remove this during refactoring.

**Comment by wladekb** [comment]
@harnash  Why did you close it?

**Comment by harnash** [comment]
@wladekb It was already merged. Those fixes should probably go in different PR.

**Pull request closed without merging**